### ISSUE TITLE
Batch multiplication in the loss file

### ISSUE THREAD

**Issue by D3lt4lph4** [pierluigiferrari/ssd_keras#185]
Hi,

Just a small question about some comments in the loss function. It is written "Keras has the annoying habit of dividing the loss by the batch size" to justify the multiplication at the end. But when I look at the code, to me, it seems like the multiplication is required because of the division by n_positive.

Correct me if I'm wrong but n_positive if the sum of all positive in the batch, thus, when dividing by n_positives you actually divide by "avg_n_positive * batch_size" and this requires the multiplication by batch_size afterwards to correct it.

Do you see any problems in this reasoning and if so where ?

Thanks,
Benjamin

**Comment by pierluigiferrari** [comment]
The reasoning is exactly as stated in the inline comment: We need to multiply by the batch size to compensate for the fact that Keras subsequently divides by it. `n_positive` ("N" in the loss function in the paper), the number of all matched boxes across the whole batch, is the number by which we want to divide.

**Comment by D3lt4lph4** [comment]
Ho, okay, got it, I read N as a per image value instead of a batch value (reading the paper again I find it to be really ambiguous regarding that part). Anyway, the comment makes sense when seeing it that way.

Thanks !
